src/screens/SignUp/index.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react'
+import NavigationService from '../../navigationService'
+import { ImageBackground, Text, View, Image, TextInput, TouchableHighlight, ActivityIndicator, KeyboardAvoidingView } from 'react-native'
+import * as firebase from 'firebase'
+import { Formik } from 'formik'
+import * as Yup from 'yup'
+import AppText from '../../helpers/AppText'
+import styles from './styles'
+
+const SignUp = () => {
+  const [authenticating, setAuthenticating] = useState(false)
+  const [error, setError] = useState(null)
+
+  const handleSubmit = async (values) => {
+    setAuthenticating(true)
+    try {
+      await firebase.auth().createUserWithEmailAndPassword(values.email, values.password)
+      NavigationService.navigate('Home')
+    } catch (error) {
+      setError(error)
+      setAuthenticating(false)
```

Comment:
My previous comment was removed by the latest commit. This statement should be in a `finally` block.

Reply:
My bad! I was pushing -f as Abel requested a rebase. There is some problem though depending on which React Native navigator is being used. If using SwitchNavigator, then setting state in the finally block induces a warning as the component will be unmounted. If using StackNavigator, no warning appears (I believe the component is still mounted, but the docs are not clear in this aspect)

Reply:
That's weird... Both `catch` and `finally` should execute one right after the other... Even before the component is unmounted... I don't know though, I'm not an expert in React Native. It may be possible for the component to be unmounted...

Reply:
Yes indeed. Might be something on the navigation. It could be that if you are navigating somewhere in the component tree, then all others (that are on different "branches") gets signaled for "WillBeUnmouted" thus giving the warning.